webapp/src/ts/services/enketo.service.ts

```diff
@@ -378,12 +387,24 @@ export class EnketoService {
       .find('.btn.previous-page')
       .off('.pagemode')
       .on('click.pagemode', () => {
-        window.history.back();
+        if (isModal) {
+          form.pages.prev();
+        } else {
+          window.history.back();
+        }
         this.forceRecalculate(form);
         return false;
       });
   }
 
+  private isFormInModal($wrapper) {
+    if (!$wrapper || !$wrapper.length) {
+      return false;
+    }
+
+    return $wrapper.closest('.modal .modal-body').length > 0;
+  }
+
```

Comment:
This seems fragile because it relies on the CSS classes in the DOM. If someone decided one day to change the names you wouldn't expect the training forms to suddenly stop working.. It would be better to pass in a boolean when rendering the form so internally this just checks the boolean.